cryptography/src/bls12381/dkg.rs

```diff
@@ -0,0 +1,1702 @@
+            })
+            .collect::<BTreeMap<_, _>>();
+        let results = reveals
+            .values()
+            .cloned()
+            .map(AckOrReveal::Reveal)
+            .collect::<Vec<_>>();
+        let priv_msgs = reveals.into_iter().collect::<Vec<_>>();
+        let commitment = Poly::commit(my_poly);
+        let pub_msg = DealerPubMsg { commitment };
+        let transcript = {
+            let t = transcript_for_round(&round_info);
+            transcript_for_dealer(&t, &me.public_key(), &pub_msg)
+        };
+        let this = Self {
+            me,
+            round_info,
+            pub_msg: pub_msg.clone(),
+            results,
+            transcript,
+        };
+        Ok((this, pub_msg, priv_msgs))
+    }
+
+    /// Process an acknowledgement from a player.
+    ///
+    /// Acknowledgements should really only be processed once per player,
+    /// but this method is idempotent nonetheless.
+    pub fn receive_player_ack(
+        &mut self,
+        player: S::PublicKey,
+        ack: PlayerAck<S::PublicKey>,
+    ) -> Result<(), Error> {
+        let index = self.round_info.player_index(&player)?;
+        if self.transcript.verify(&player, &ack.sig) {
+            self.results[index as usize] = AckOrReveal::Ack(ack);
+        }
+        Ok(())
+    }
+
+    /// Finalize the dealer, producing a signed log.
+    ///
+    /// This should be called at the point where no more acks will be processed.
+    pub fn finalize(self) -> SignedDealerLog<V, S> {
+        let log = DealerLog {
+            pub_msg: self.pub_msg,
+            results: self.results,
+        };
+        SignedDealerLog::sign(&self.me, &self.round_info, log)
+    }
+}
+
+#[allow(clippy::type_complexity)]
+fn select<V: Variant, P: PublicKey>(
+    round_info: &Info<V, P>,
+    logs: BTreeMap<P, DealerLog<V, P>>,
+) -> Result<Vec<(P, DealerLog<V, P>)>, Error> {
+    let required_commitments = round_info.required_commitments() as usize;
+    let transcript = transcript_for_round(round_info);
+    let out = logs
+        .into_iter()
+        .filter_map(|(dealer, log)| {
+            round_info.dealer_index(&dealer).ok()?;
+            if !round_info.check_dealer_pub_msg(&dealer, &log.pub_msg) {
+                return None;
+            }
+            let results_iter = log.zip_players(&round_info.players)?;
+            let transcript = transcript_for_dealer(&transcript, &dealer, &log.pub_msg);
+            let mut reveal_count = 0;
+            let max_reveals = round_info.max_reveals();
+            for (player, result) in results_iter {
+                match result {
+                    AckOrReveal::Ack(ack) => {
+                        if !transcript.verify(player, &ack.sig) {
+                            return None;
+                        }
+                    }
+                    AckOrReveal::Reveal(priv_msg) => {
+                        reveal_count += 1;
+                        if reveal_count > max_reveals {
```

Comment:
Confirming we don't attempt to post a commitment if we exceed this? If so, can we add some test for this?

Reply:
Modified the behavior here to post an empty log if there are too many reveals, and added tests for this.

Reply:
Added a comment to just post nothing in that case (rather than an empty log) -> I don't like the idea of including but ignoring certain data.